azurelinuxagent/ga/cgroupconfigurator.py

```diff
@@ -198,6 +198,11 @@ def initialize(self):
                 log_cgroup_info('Agent cgroups enabled: {0}'.format(self._agent_cgroups_enabled))
                 self._initialized = True
 
+                if self._cgroups_api is not None and not self._cgroups_api.can_enforce_cpu():
```

Comment:
in what scenario do we need to do this additional clean up?

Reply:
This is needed for fixing the current issue. In next release, we enable cgroups v2 for resource monitoring but not the enforcement. This check needed to reset that old-agent quota

Reply:
I understand that it's kind of confusing having cleanups at multiple places, but we have different scarious to handle